_posts/2020-12-04-release-1.12.0.md

```diff
@@ -0,0 +1,313 @@
+
+<div class="alert alert-info small" markdown="1">
+<b>Note:</b> This does not mean that the ZooKeeper dependency will be dropped, just that there will be an alternative for users of Flink on Kubernetes.
+</div>
+
+<hr>
+
+### Other Improvements
+
+**Migration of existing connectors to the new Data Source API**
+
+The previous release introduced a new Data Source API ([FLIP-27](https://cwiki.apache.org/confluence/display/FLINK/FLIP-27%3A+Refactor+Source+Interface)), allowing to implement connectors that work both as bounded (batch) and unbounded (streaming) sources. In Flink 1.12, the community started porting existing source connectors to the new interfaces, starting with the FileSystem connector ([FLINK-19161](https://issues.apache.org/jira/browse/FLINK-19161)). 
+
+<div class="alert alert-danger small" markdown="1">
+<b>Attention:</b> The unified source implementations will be completely separate connectors that are not snapshot-compatible with their legacy counterparts.
+</div>
+
+**Pipelined Region Scheduling ([FLIP-119](https://cwiki.apache.org/confluence/display/FLINK/FLIP-119+Pipelined+Region+Scheduling#FLIP119PipelinedRegionScheduling-BulkSlotAllocation))**
+
+Flink’s scheduler has been largely designed to address batch and streaming workloads separately. This release introduces a **unified** scheduling strategy that identifies sets of tasks connected via blocking data exchanges to break down the execution graph into _pipelined regions_. This allows to schedule each region only when there’s data to perform work and only deploy it once all the required resources are available; as well as to restart failed regions independently. In particular for batch jobs, the new strategy leads to more efficient resource utilization and eliminates deadlocks.
+
+**Support for Sort-Merge Blocking Shuffles ([FLIP-148](https://cwiki.apache.org/confluence/display/FLINK/FLIP-148%3A+Introduce+Sort-Merge+Based+Blocking+Shuffle+to+Flink))**
+
+To improve the stability, performance and resource utilization of large-scale batch jobs, the community introduced sort-merge blocking shuffle as an alternative to the hash-based implementation that Flink already used. This approach can reduce shuffle time [significantly](https://www.mail-archive.com/dev@flink.apache.org/msg42472.html), and uses fewer file handles and file write buffers (which is problematic for large-scale jobs). Further optimizations will be implemented in upcoming releases ([FLINK-19614](https://issues.apache.org/jira/browse/FLINK-19614)).
```

Comment:
I would also drop the term "hash-based shuffle", because I think it is not really correct. The data partitioning is the same for both (hash-based), just the mapping of partitions to files is different (1:1 in the original shuffle, sort-merge-based clustering in the new implementation).

For this announcement, I would just refer to it as the "original shuffle" (or "original file-per-partition shuffle") versus "sort-merge shuffle".